### PR TITLE
ObjectTree/List: Add ariaRootRole option.  Addresses: #65939

### PR DESCRIPTION
**Pull request by cleidigh**
@joaomoreno 
@roblourens 

Addresses:  #65939

- Add new enum EListAriaRootRole for ariaRootRole option (mixed feelings about 'E' prefix)

```ts
// Default tree/list root role 'tree' does not support interactive element labeling, 'form' does
export enum EListAriaRootRole {
	TREE = 'tree',		// default tree structure role
	FORM = 'form'		// unstructured - allows interactive elements within tree/list structure
}
```

- Use EListAriaRootRole.TREE as default role, no need to change other clients
- Update settingsTree/settingsEditor2 to use EListAriaRootRole.FORM
- Handle the two valid roles in listWidget
- Remove all aria attribute manipulation in Settings Editor

Note:

Reader behavior related to this issue has changed with different versions of readers, Electron
and Chromium.  It it appears from testing (both NVDA & Voiceover) that the readers
will ignore the tree related attributes if the root role != 'tree'.  I therefore decided
NOT to alter any application logic for those attributes: item role, aria-level, aria-posinset, aria-setsize

I think this is the conservative approach and the minimal amount of changes. I could see 
needing to change the item role in the future possibly, but for now works.

Tested:
- Windows 10, NVDA 2018.4.1 - latest
- OS X Mojave, Voiceover

### CODE REVIEW

File: src/vs/base/browser/ui/list/listWidget.ts [hunk 1]
```diff
@@ -779,11 +779,13 @@ export class DefaultStyleController implements IStyleController {
 	}
 }
 
+
 export interface IListOptions<T> extends IListStyles {
 	readonly identityProvider?: IIdentityProvider<T>;
 	readonly dnd?: IListDragAndDrop<T>;
 	readonly enableKeyboardNavigation?: boolean;
 	readonly keyboardNavigationLabelProvider?: IKeyboardNavigationLabelProvider<T>;
+	readonly ariaRootRole?: EListAriaRootRole;
```
Comment (joaomoreno) [inline src/vs/base/browser/ui/list/listWidget.ts]: You can just call it `ariaRole`.

File: src/vs/base/browser/ui/list/list.ts [hunk 2]
```diff
@@ -62,6 +62,12 @@ export interface IIdentityProvider<T> {
 	getId(element: T): { toString(): string; };
 }
 
+// Default tree/list root role 'tree' does not support interactive element labeling, 'form' does
+export enum EListAriaRootRole {
```
Comment (joaomoreno) [inline src/vs/base/browser/ui/list/list.ts]: `ListAriaRole` would be enough here.

**Review by joaomoreno — CHANGES_REQUESTED** [review]
@cleidigh Love this PR. I'm glad you convinced me about your authorship of this. I have only minor change requests. Leaving the rest to @roblourens

**Comment by cleidigh** [comment]
> @cleidigh Love this PR. I'm glad you convinced me about your authorship of this. I have only minor change requests. Leaving the rest to @roblourens

Thanks a lot @joaomoreno I really enjoy working on this with you guys especially for such a great project !
As I have offered to @bpasero  & @roblourens if you guys have any special projects, issues or even experimental research, I would be glad to give it a try.  I like getting deeper into the CodeBase 🔢

File: src/vs/base/browser/ui/list/list.ts (hunk 2 above)
Comment (cleidigh) [reply src/vs/base/browser/ui/list/list.ts]: 👍

File: src/vs/base/browser/ui/list/listWidget.ts (hunk 1 above)
Comment (cleidigh) [reply src/vs/base/browser/ui/list/listWidget.ts]: 👍

File: src/vs/workbench/parts/preferences/browser/settingsTree.ts [hunk 3]
```diff
@@ -1292,11 +1287,14 @@ export class SettingsTree extends ObjectTree<SettingsTreeElement> {
 	) {
 		const treeClass = 'settings-editor-tree';
 
+		// Must use role = EListAriaRootRole.FORM for interactive elements in settingsTree
```
Comment (roblourens) [inline src/vs/workbench/parts/preferences/browser/settingsTree.ts]: I think this comment isn't necessary

**Review by roblourens — APPROVED** [review]
Thanks! Looks good to me

File: src/vs/workbench/parts/preferences/browser/settingsTree.ts (hunk 3 above)
Comment (roblourens) [reply src/vs/workbench/parts/preferences/browser/settingsTree.ts]: I'll just merge and delete it :)

**Comment by cleidigh** [comment]
Thanks Guys

**Comment by dbaeumer** [comment]
Verified in code.

**Pull request merged**